### PR TITLE
Chore: Add extension generators to Generate Model script

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

+ Add missing `kubernetes-model-flowcontrol` to generateModel script
+ With this change now extension model will be built too in Go Build github workflow and verified for any schema changes.
+ Remove dependency `org.skyscraper:jsonassert` I couldn't find any place where it's being used.
+ Remove plugin `com.mysema.maven:apt-maven-plugin` from extensions. I didn't find its usages in other extensions and kubernetes-model-generator
+ ~~generateModel script only runs `make gobuild`. No need to run `mvn clean install` too~~

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by manusa** [comment]
> generateModel script only runs make gobuild. No need to run mvn clean install too

**That's there on purpose.** (#3079)

image


The whole point is to see if any of the generated classes gets altered. This can happen in two ways:
1 - By altering the schema, so yes this is what you're verifying with the JSON
2 - By changing any of the dependencies or classes that transform the json schema into Java classes.

In addition, the generateModel.sh or generateModelDocker.sh script is not only used to verify no schema is changed (in CI), but should be invoked whenever go files are modified to update the schemas and Java types.

File: kubernetes-model-generator/generateModel.sh [hunk 1]
```diff
@@ -74,6 +88,7 @@ echo "Installing required common modules"
 mvn clean install -f ../pom.xml -N
 mvn clean install -N
 mvn clean install -pl kubernetes-model-common -pl kubernetes-model-jsonschema2pojo
+cd ../model-annotator && mvn clean install && cd -
```
Comment (manusa) [inline kubernetes-model-generator/generateModel.sh:91]: Could you use `mvn clean install -f ../model-annotator/pom.xml` for consistency?

File: kubernetes-model-generator/generateModel.sh [hunk 2]
```diff
@@ -74,6 +88,11 @@ echo "Installing required common modules"
 mvn clean install -f ../pom.xml -N
 mvn clean install -N
 mvn clean install -pl kubernetes-model-common -pl kubernetes-model-jsonschema2pojo
+mvn clean install -DskipTests 
+mvn clean install -f ../model-annotator/pom.xml
+mvn clean install -DskipTests -f ../kubernetes-client/pom.xml
+mvn clean install -DskipTests -f ../kubernetes-server-mock/pom.xml
+mvn clean install -DskipTests -f ../extensions/
```
Comment (manusa) [inline kubernetes-model-generator/generateModel.sh:95]: Only the annotator, or required dependencies should be compiled.

None of the other modules should be necessary. Models should only depend on models (and annotators), if some other dependency is required, there must be some error in the pom.xml definition of the affected extension.

File: .github/workflows/gobuild.yml [hunk 3]
```diff
@@ -35,7 +35,7 @@ jobs:
       - name: Check No Schema file modified
         run: |
           git add .
-          MODIFIED_KUBE_SCHEMA=`git status | grep 'kube-schema.json\|src/generated/java' | wc -l`
+          MODIFIED_KUBE_SCHEMA=`git status | grep -v 'validation-schema.json' | grep -E '*-schema.json|src/generated/java' | wc -l`
```
Comment (manusa) [inline .github/workflows/gobuild.yml:38]: I'm not sure if the `*` in the regex there is actually doing anything

File: kubernetes-model-generator/generateModel.sh [hunk 4]
```diff
@@ -53,8 +54,40 @@ declare -a modules=(
...
+declare -a extensionModules=(
+    "../extensions/knative/generator"
+    "../extensions/camel-k/generator-v1"
+    "../extensions/camel-k/generator-v1alpha1"
+    "../extensions/chaosmesh/generator"
+    "../extensions/service-catalog/generator"
+    "../extensions/tekton/generator-v1alpha1"
+    "../extensions/tekton/generator-v1beta1"
+    "../extensions/volumesnapshot/generator"
+)
```
Comment (manusa) [inline kubernetes-model-generator/generateModel.sh:78]: I'd rather keep things simpler. Why not declare those together with the rest of modules?

File: kubernetes-model-generator/generateModel.sh [hunk 5]
```diff
@@ -53,8 +54,40 @@ declare -a modules=(
...
+
 generateAll() {
-  for module in ${modules[*]}
+  echo "Compiling core modules"
+  generateSetOfModules "${modules[@]}"
+  echo "Compiling extensions"
+  extensionInstallCommonModules
```
Comment (manusa) [inline kubernetes-model-generator/generateModel.sh:84]: Why isn't this run with the rest of prerequisites (after line `mvn clean install -pl kubernetes-model-common -pl kubernetes-model-jsonschema2pojo`)?

File: kubernetes-model-generator/generateModel.sh [hunk 6]
```diff
@@ -53,8 +54,40 @@ declare -a modules=(
...
+  generateSetOfModules "${extensionModules[@]}"
```
Comment (manusa) [inline kubernetes-model-generator/generateModel.sh:85]: https://github.com/fabric8io/kubernetes-client/pull/3203/files#r662287448

File: kubernetes-model-generator/generateModel.sh [hunk 7]
```diff
@@ -52,9 +53,34 @@ declare -a modules=(
...
+
+declare -a extensionModuleParents=(
+    "../extensions/knative/pom.xml"
+    "../extensions/camel-k/pom.xml"
+    "../extensions/camel-k/pom.xml"
+    "../extensions/chaosmesh/pom.xml"
+    "../extensions/service-catalog/pom.xml"
+    "../extensions/tekton/pom.xml"
+    "../extensions/tekton/pom.xml"
+    "../extensions/volumesnapshot/pom.xml"
```
Comment (manusa) [inline kubernetes-model-generator/generateModel.sh:74]: Remove duplicates

**Review by manusa — APPROVED** [review]

**Pull request merged**